sky/metrics/utils.py

```diff
@@ -56,10 +210,16 @@ def start_svc_port_forward(context: str, namespace: str, service: str,
 
         # read output line by line to find the local port
         if port_forward_process.stdout:
-            line = port_forward_process.stdout.readline()
-            if line:
-                # look for 'Forwarding from 127.0.0.1:XXXXX -> service_port'
-                match = re.search(r'Forwarding from 127\.0\.0\.1:(\d+)', line)
+            # Wait up to 1s for data to be available without blocking
+            r, _, _ = select.select([port_forward_process.stdout], [], [],
```

Comment:
`start_svc_port_forward` can block the event loop, can we run it in a separate thread?

Reply:
Updated